File: 2023-2024/sourcecode/mod3/lab8/LiveDataBasicsJava/app/src/main/java/com/example/livedatabasics_java/ui/fragment/CounterFragment.java
package com.example.livedatabasics_java.ui.fragment;

import android.annotation.SuppressLint;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.livedatabasics_java.R;
import com.example.livedatabasics_java.databinding.FragmentCounterBinding;
import com.example.livedatabasics_java.viewmodel.CounterViewModel;

public class CounterFragment extends Fragment {

    private FragmentCounterBinding binding;

    private CounterViewModel viewModel;

    @SuppressLint("SetTextI18n")
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentCounterBinding.inflate(inflater);

        viewModel = new ViewModelProvider(requireActivity()).get(CounterViewModel.class);

        viewModel.getCounter().observe(getViewLifecycleOwner(),
                newValue -> binding.showCount.setText(newValue.toString()));

        binding.increaseButton.setOnClickListener(view -> viewModel.increase());
        binding.decreaseButton.setOnClickListener( view -> viewModel.decrease());
        binding.resetButtton.setOnClickListener (view -> viewModel.clear());

        return binding.getRoot();
    }
}

File: 2023-2024/sourcecode/mod3/lab8/LiveDataBasicsKotlin/app/src/main/java/com/example/livedatabasics_kotlin/ui/fragment/CounterFragment.kt
package com.example.livedatabasics_kotlin.ui.fragment

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.viewModels
import com.example.livedatabasics_kotlin.databinding.FragmentCounterBinding
import com.example.livedatabasics_kotlin.viewmodel.CounterViewModel

class CounterFragment : Fragment() {

    private lateinit var binding: FragmentCounterBinding

    private val viewModel: CounterViewModel by viewModels()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentCounterBinding.inflate(inflater)

        viewModel.counter.observe(viewLifecycleOwner) { newValue ->
            binding.showCount.text = newValue.toString()
        }

        binding.increaseButton.setOnClickListener { viewModel.increase() }
        binding.decreaseButton.setOnClickListener { viewModel.decrease() }
        binding.resetButtton.setOnClickListener { viewModel.clear() }

        return binding.root
    }
}